From 3215ec61d11de1f512889119a1d4ec4b0554c88c Mon Sep 17 00:00:00 2001
From: "Guoniu.Zhou" <guoniu.zhou@nxp.com>
Date: Mon, 19 Nov 2018 15:14:05 +0800
Subject: [PATCH 5151/5242] MLK-20404: sensor: fix potential memory leaks for
 ov5640

commit  cf141997953ee22c429cac3f49067ede7498969d from
https://source.codeaurora.org/external/imx/linux-imx.git

Driver does not free memory when it's removed or encounter
an error in probe. So fix potential memory leaks.

Signed-off-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
(cherry picked from commit c236a94088903088f8ac32b12764b6d63ada43ea)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/media/platform/imx8/ov5640_mipi_v3.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/media/platform/imx8/ov5640_mipi_v3.c b/drivers/media/platform/imx8/ov5640_mipi_v3.c
index 02a3db4..e1197e7 100644
--- a/drivers/media/platform/imx8/ov5640_mipi_v3.c
+++ b/drivers/media/platform/imx8/ov5640_mipi_v3.c
@@ -1201,7 +1201,9 @@ static int ov5640_probe(struct i2c_client *client,
 	u8 chip_id_high, chip_id_low;
 	struct ov5640 *sensor;
 
-	sensor = kmalloc(sizeof(*sensor), GFP_KERNEL);
+	sensor = devm_kmalloc(dev, sizeof(*sensor), GFP_KERNEL);
+	if (!sensor)
+		return -ENOMEM;
 	/* Set initial values for the sensor struct. */
 	memset(sensor, 0, sizeof(*sensor));
 
-- 
1.7.9.5

